tests_metricflow/integration/test_configured_cases.py

```diff
@@ -361,3 +362,16 @@ def test_case(
     )
     # If we sort, it's effectively not checking the order whatever order that the output was would be overwritten.
     assert_dataframes_equal(actual, expected, sort_columns=not case.check_order, allow_empty=case.allow_empty)
+
+    # Check that the parse result and the dataflow plan show the same semantic models read.
+    if name in {"itest_dimensions.yaml/distinct_values_query_with_metric_filter"}:
```

Comment:
@plypaul did some digging and it turns out this bug is not in the metric filters feature, it's in queries without metrics in general. You can see [this test case](https://github.com/dbt-labs/metricflow/pull/1209/files) gets the same error. I'll put up an issue to fix that, but you might want to update the message below to reflect that.